subsys/net/lib/nrf_cloud/include/nrf_cloud_fota.h

```diff
@@ -102,19 +106,26 @@ typedef void (*nrf_cloud_fota_callback_t)
 typedef void (*nrf_cloud_fota_ble_callback_t)
 	(const struct nrf_cloud_fota_ble_job * const ble_job);
 
+struct nrf_cloud_fota_init_param {
+	nrf_cloud_fota_callback_t evt_cb;
+#if defined(CONFIG_NRF_CLOUD_FOTA_SMP)
```

Comment:
Or perhaps a better question: If it is possible, would it be detrimental to do so?

Reply:
i could remove the `if defined` and make this a `void *` with the comment stating that it should be of the type `dfu_target_reset_cb_t`?

Reply:
I don't think you even would need to do that, since `dfu_target_reset_cb_t` is always available

You could just document that it is ignored if `CONFIG_NRF_CLOUD_FOTA_SMP` is disabled. Maybe add an error print if the user provides a value when `CONFIG_NRF_CLOUD_FOTA_SMP` is enabled

But it's not a hard-line issue for me, especially if completing this is urgent

Reply:
dfu_target_reset_cb_t requires `mcumgr_smp_client.h` to be included... which produces a build error if cbor is not enabled...
anyway, i just made it `void*`